commit 265376282a58c71f1ec9215653e98c602524b91c
Author: Cameron Nemo <cnemo@tutanota.com>
Date:   Mon Sep 7 08:37:24 2020 -0700

    rockchip: Pinebook Pro: fix mmc boot ordering
    
    On the Pinebook Pro, and perhaps some other boards, the mmc indices are
    opposite from what is implied by the comment in
    include/configs/rockchip-common.h.
    
    This commit fixes the boot ordering to prioritize the SD card over eMMC.
    
    Most distributions of U-Boot that actually target rk3399 boards apply a
    similar patch, such as Manjaro (shipped by default on PBP) and armbian.
    
    https://gitlab.manjaro.org/manjaro-arm/packages/core/uboot-pinebookpro/-/blob/60ed56b9bf8809309ec6252bfc45bca38c2b858a/0002-Correct-boot-order-to-be-USB-SD-eMMC.patch
    https://github.com/armbian/build/blob/5a2b2c360b9c26ca4bd0d309af7cd3994fd08b7d/patch/u-boot/u-boot-rockchip64-mainline/general-prioritize-sd.patch
    
    Signed-off-by: Cameron Nemo <cnemo@tutanota.com>

diff --git include/configs/pinebook-pro-rk3399.h include/configs/pinebook-pro-rk3399.h
index d478b19917..59780bfb0b 100644
--- include/configs/pinebook-pro-rk3399.h
+++ include/configs/pinebook-pro-rk3399.h
@@ -14,6 +14,14 @@
 
 #include <configs/rk3399_common.h>
 
+/* On Pinebook Pro, mmc1 is SD and mmc0 is eMMC */
+#if CONFIG_IS_ENABLED(CMD_MMC)
+	#undef BOOT_TARGET_MMC
+	#define BOOT_TARGET_MMC(func) \
+		func(MMC, mmc, 1) \
+		func(MMC, mmc, 0)
+#endif
+
 #define SDRAM_BANK_SIZE			(2UL << 30)
 
 #define CONFIG_USB_OHCI_NEW
